src/ModelContextProtocol/Client/McpClientFactory.cs

```diff
@@ -62,78 +51,15 @@ public static async Task<IMcpClient> CreateAsync(
         }
         catch
         {
-            if (transport is IAsyncDisposable asyncDisposableTransport)
+            if (clientTransport is IAsyncDisposable asyncDisposableTransport)
```

Comment:
If memory serves, the transport provided to McpServerFactory doesn't have its ownership transferred to the McpServer, such that the McpServer's disposal doesn't dispose of the server transport. But here the client transport is being disposed as part of the client. Are we purposefully different between the two?

Reply:
Good call. This catch was only disposing the session-level ITransport because it was newly created by the createTransportFunc, and the caller never got a reference to it or a chance to dispose it when CreateAsync throws.

I don't think it makes sense to do this on the IClientTransport. It'd be nice if that were reusable, and not just when CreateAsync doesn't throw.

I think we can remove the outer try/catch for the transport disposal entirely now.

Reply:
Agreed, the `IClientTransport` implementations are mostly stateless today (with the sole exception of `SseClientTransport` instances that define their own `HttpClient` instance, disposing which here does not seem appropriate).